airflow/models/base_reschedule_poke_operator.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Base Operator for kicking off a long running
+operations and polling for completion with reschedule mode.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Dict, List, Iterable, Optional, Union
+from time import sleep
+from datetime import timedelta
+
+from airflow.exceptions import AirflowException, AirflowSensorTimeout, \
+    AirflowSkipException, AirflowRescheduleException
+from airflow.models import SkipMixin, TaskReschedule
+from airflow.models.xcom import XCOM_EXTERNAL_RESOURCE_ID_KEY
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+from airflow.utils import timezone
+from airflow.utils.decorators import apply_defaults
+from airflow.ti_deps.deps.ready_to_reschedule import ReadyToRescheduleDep
+
+
+class BaseReschedulePokeOperator(BaseOperator, SkipMixin, ABC):
```

Comment:
I wonder how `on_kill` get invoked if we shutdown the task through Airflow UI when the task is waiting to be rescheduled.

Reply:
that is a really good point. I'm not sure. I imagine it wouldn't run until the task was rescheduled.